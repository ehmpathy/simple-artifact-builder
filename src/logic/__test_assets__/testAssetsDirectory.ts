export const TEST_ASSETS_DIRECTORY = __dirname;
