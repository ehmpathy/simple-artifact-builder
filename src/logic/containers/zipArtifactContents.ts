import archiver from 'archiver';
import fs from 'fs';
import { assertThatArtifactContentsWereBuilt } from '../contents/assetThatArtifactContentsWereBuilt';
import { UserInputError } from '../UserInputError';

/**
 * zips a directory
 *
 * ref: https://stackoverflow.com/a/51518100/3068233
 */
const zipDirectory = ({ source, destination }: { source: string; destination: string }) => {
  const archive = archiver('zip', { zlib: { level: 9 } });
  const stream = fs.createWriteStream(destination);
  return new Promise((resolve, reject) => {
    archive
      .directory(source, false)
      .on('error', (err) => reject(err))
      .pipe(stream);
    stream.on('close', () => resolve(true));
    archive.finalize();
  });
};

export const zipArtifactContents = async ({ projectRootDirectory }: { projectRootDirectory: string }) => {
  const toAbsolutePath = (relativePath: string) => `${projectRootDirectory}/${relativePath}`;

  // check that artifact contents exist
  await assertThatArtifactContentsWereBuilt({ projectRootDirectory });

  // zip the contents
  await zipDirectory({
    source: toAbsolutePath('.artifact/contents'),
    destination: toAbsolutePath('.artifact/contents.zip'),
  });
};
