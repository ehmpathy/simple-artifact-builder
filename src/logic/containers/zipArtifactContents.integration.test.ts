import fs from 'fs';
import StreamZip from 'node-stream-zip';

import { zipArtifactContents } from './zipArtifactContents';
import { TEST_ASSETS_DIRECTORY } from '../__test_assets__/testAssetsDirectory';

describe('zipArtifactContents', () => {
  it('should be able to zip artifact contents of built artifact', async () => {
    const projectRootDirectory = `${TEST_ASSETS_DIRECTORY}/nextjs-mui-project-next-dir-for-zipping`;
    const targetZipFilePath = `${projectRootDirectory}/.artifact/contents.zip`;

    // remove the previous file if it exists
    await fs.promises.unlink(targetZipFilePath).catch(() => {}); // do nothing on error, e.g., if file didn't already exist

    // zip the contents
    await zipArtifactContents({ projectRootDirectory });

    // check that the file was zipped
    await new Promise((resolve) => {
      const zip = new StreamZip({
        file: targetZipFilePath,
        storeEntries: true,
      });
      zip.on('ready', () => {
        expect(zip.entriesCount).toBeGreaterThan(10); // should have more than 10, sanity check
        expect(zip.entries()).toMatchSnapshot();
        zip.close(); // close the file once done
        resolve(true); // and end the promise
      });
    });
  });
});
