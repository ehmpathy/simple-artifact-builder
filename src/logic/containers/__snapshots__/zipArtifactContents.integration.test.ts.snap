// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zipArtifactContents should be able to zip artifact contents of built artifact 1`] = `
Object {
  ".next/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 6,
    "headerOffset": 234816,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": ".next/",
    "offset": 0,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  ".next/BUILD_ID": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 23,
    "crc": 890034749,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 14,
    "headerOffset": 234868,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/BUILD_ID",
    "offset": 36,
    "size": 21,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/build-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 389,
    "crc": 2251008148,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 25,
    "headerOffset": 234928,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/build-manifest.json",
    "offset": 119,
    "size": 1417,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/export-marker.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 78,
    "crc": 273904047,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 24,
    "headerOffset": 234999,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/export-marker.json",
    "offset": 579,
    "size": 94,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/images-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 197,
    "crc": 496172047,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 26,
    "headerOffset": 235069,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/images-manifest.json",
    "offset": 727,
    "size": 321,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/next-server.js.nft.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1885,
    "crc": 3379965061,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 29,
    "headerOffset": 235141,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/next-server.js.nft.json",
    "offset": 996,
    "size": 13395,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/prerender-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 216,
    "crc": 3519755419,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 29,
    "headerOffset": 235216,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/prerender-manifest.json",
    "offset": 2956,
    "size": 312,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/react-loadable-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 4,
    "crc": 2745614147,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 34,
    "headerOffset": 235291,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/react-loadable-manifest.json",
    "offset": 3247,
    "size": 2,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/required-server-files.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1021,
    "crc": 4139736472,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 32,
    "headerOffset": 235371,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/required-server-files.json",
    "offset": 3331,
    "size": 2020,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/routes-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 222,
    "crc": 1833545114,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 26,
    "headerOffset": 235449,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/routes-manifest.json",
    "offset": 4430,
    "size": 423,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 13,
    "headerOffset": 235521,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": ".next/server/",
    "offset": 4724,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  ".next/server/chunks/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 20,
    "headerOffset": 235580,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": ".next/server/chunks/",
    "offset": 4767,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  ".next/server/chunks/600.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1966,
    "crc": 741037061,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 26,
    "headerOffset": 235646,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/chunks/600.js",
    "offset": 4817,
    "size": 11653,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/chunks/664.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 17615,
    "crc": 598123653,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 26,
    "headerOffset": 235718,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/chunks/664.js",
    "offset": 6855,
    "size": 70910,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/chunks/730.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 513,
    "crc": 3846672780,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 26,
    "headerOffset": 235790,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/chunks/730.js",
    "offset": 24542,
    "size": 1326,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/chunks/859.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 8453,
    "crc": 1781199750,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 26,
    "headerOffset": 235862,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/chunks/859.js",
    "offset": 25127,
    "size": 40411,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/chunks/font-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 724,
    "crc": 4242774748,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 38,
    "headerOffset": 235934,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/chunks/font-manifest.json",
    "offset": 33652,
    "size": 8243,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/font-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 724,
    "crc": 4242774748,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 31,
    "headerOffset": 236018,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/font-manifest.json",
    "offset": 34460,
    "size": 8243,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/middleware-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 61,
    "crc": 1206023686,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 37,
    "headerOffset": 236095,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/middleware-manifest.json",
    "offset": 35261,
    "size": 84,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages-manifest.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 90,
    "crc": 1083429373,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 32,
    "headerOffset": 237176,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages-manifest.json",
    "offset": 62634,
    "size": 187,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 19,
    "headerOffset": 236178,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": ".next/server/pages/",
    "offset": 35405,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  ".next/server/pages/404.html": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1920,
    "crc": 737314639,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 27,
    "headerOffset": 236243,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/404.html",
    "offset": 35454,
    "size": 10992,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/500.html": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1916,
    "crc": 3122883166,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 27,
    "headerOffset": 236316,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/500.html",
    "offset": 37447,
    "size": 10978,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/_app.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1291,
    "crc": 3026014738,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 26,
    "headerOffset": 236389,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/_app.js",
    "offset": 39436,
    "size": 5097,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/_app.js.nft.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 2492,
    "crc": 95232759,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 35,
    "headerOffset": 236461,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/_app.js.nft.json",
    "offset": 40799,
    "size": 20747,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/_document.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 2050,
    "crc": 3720781039,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 31,
    "headerOffset": 236542,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/_document.js",
    "offset": 43372,
    "size": 7151,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/_document.js.nft.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 3285,
    "crc": 3460210625,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 40,
    "headerOffset": 236619,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/_document.js.nft.json",
    "offset": 45499,
    "size": 28399,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/_error.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1381,
    "crc": 1896188353,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 28,
    "headerOffset": 236705,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/_error.js",
    "offset": 48870,
    "size": 3830,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/_error.js.nft.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 184,
    "crc": 3559690963,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 37,
    "headerOffset": 236779,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/_error.js.nft.json",
    "offset": 50325,
    "size": 645,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/about.html": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 3149,
    "crc": 3770194944,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 29,
    "headerOffset": 236862,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/about.html",
    "offset": 50592,
    "size": 15733,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/about.js.nft.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 3103,
    "crc": 3207580936,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 36,
    "headerOffset": 236937,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/about.js.nft.json",
    "offset": 53816,
    "size": 27473,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/index.html": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 2623,
    "crc": 343853492,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 29,
    "headerOffset": 237019,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/index.html",
    "offset": 57001,
    "size": 13447,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/pages/index.js.nft.json": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 2853,
    "crc": 869776788,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 36,
    "headerOffset": 237094,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/pages/index.js.nft.json",
    "offset": 59699,
    "size": 22987,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/server/webpack-runtime.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1547,
    "crc": 623077852,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 31,
    "headerOffset": 237254,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/server/webpack-runtime.js",
    "offset": 62802,
    "size": 6062,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 13,
    "headerOffset": 237331,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": ".next/static/",
    "offset": 64426,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  ".next/static/chunks/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 20,
    "headerOffset": 237390,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": ".next/static/chunks/",
    "offset": 64469,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  ".next/static/chunks/31-ea2e06847afa5ee5.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 5867,
    "crc": 2298997954,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 42,
    "headerOffset": 237456,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/31-ea2e06847afa5ee5.js",
    "offset": 64519,
    "size": 17904,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/755-1f25bec492800030.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 10974,
    "crc": 1953576167,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 43,
    "headerOffset": 237544,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/755-1f25bec492800030.js",
    "offset": 70474,
    "size": 30519,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/framework-bbce3cdc1a611f18.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 42370,
    "crc": 438585328,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 49,
    "headerOffset": 237633,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/framework-bbce3cdc1a611f18.js",
    "offset": 81537,
    "size": 131009,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/main-e7f9cd3517807e31.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 27795,
    "crc": 1611452423,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 44,
    "headerOffset": 237728,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/main-e7f9cd3517807e31.js",
    "offset": 124002,
    "size": 95171,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/pages/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 26,
    "headerOffset": 237818,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": ".next/static/chunks/pages/",
    "offset": 151887,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  ".next/static/chunks/pages/_app-02318ea03113095b.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 14413,
    "crc": 3218679625,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 50,
    "headerOffset": 237890,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/pages/_app-02318ea03113095b.js",
    "offset": 151943,
    "size": 38569,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/pages/_error-2280fa386d040b66.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 176,
    "crc": 4283694030,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 52,
    "headerOffset": 237986,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/pages/_error-2280fa386d040b66.js",
    "offset": 166452,
    "size": 267,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/pages/about-a0e709c255002672.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1829,
    "crc": 577131880,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 51,
    "headerOffset": 238084,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/pages/about-a0e709c255002672.js",
    "offset": 166726,
    "size": 4617,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/pages/index-259f2ab4bd6663c8.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 1807,
    "crc": 1331114467,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 51,
    "headerOffset": 238181,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/pages/index-259f2ab4bd6663c8.js",
    "offset": 168652,
    "size": 4574,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/polyfills-a40ef1678bae11e696dba45124eadd70.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 31004,
    "crc": 1380411922,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 65,
    "headerOffset": 238278,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/polyfills-a40ef1678bae11e696dba45124eadd70.js",
    "offset": 170556,
    "size": 91434,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/chunks/webpack-514908bffb652963.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 752,
    "crc": 405297075,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 47,
    "headerOffset": 238389,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/chunks/webpack-514908bffb652963.js",
    "offset": 201671,
    "size": 1472,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/lwiVzzFlheitLtrmf0quQ/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 35,
    "headerOffset": 238482,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": ".next/static/lwiVzzFlheitLtrmf0quQ/",
    "offset": 202516,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  ".next/static/lwiVzzFlheitLtrmf0quQ/_buildManifest.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 267,
    "crc": 1784068793,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 52,
    "headerOffset": 238563,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/lwiVzzFlheitLtrmf0quQ/_buildManifest.js",
    "offset": 202581,
    "size": 453,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/lwiVzzFlheitLtrmf0quQ/_middlewareManifest.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 44,
    "crc": 2590440933,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 57,
    "headerOffset": 238661,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/lwiVzzFlheitLtrmf0quQ/_middlewareManifest.js",
    "offset": 202946,
    "size": 92,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/static/lwiVzzFlheitLtrmf0quQ/_ssgManifest.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 43,
    "crc": 1347465868,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 50,
    "headerOffset": 238764,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/static/lwiVzzFlheitLtrmf0quQ/_ssgManifest.js",
    "offset": 203093,
    "size": 77,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  ".next/trace": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 31072,
    "crc": 470693754,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 11,
    "headerOffset": 238860,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": ".next/trace",
    "offset": 203232,
    "size": 224537,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  "dist/": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 0,
    "crc": 0,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 0,
    "fnameLen": 5,
    "headerOffset": 238917,
    "inattr": 0,
    "isDirectory": true,
    "method": 0,
    "name": "dist/",
    "offset": 234361,
    "size": 0,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 10,
  },
  "dist/handler.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 161,
    "crc": 2679258422,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 15,
    "headerOffset": 238968,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": "dist/handler.js",
    "offset": 234396,
    "size": 259,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
  "dist/server.js": Object {
    "attr": "__REDACTED__",
    "comLen": 0,
    "comment": null,
    "compressedSize": 138,
    "crc": 3062844332,
    "diskStart": 0,
    "extraLen": 0,
    "flags": 8,
    "fnameLen": 14,
    "headerOffset": 239029,
    "inattr": 0,
    "isDirectory": false,
    "method": 8,
    "name": "dist/server.js",
    "offset": 234618,
    "size": 188,
    "time": "__REDACTED__",
    "verMade": 813,
    "version": 20,
  },
}
`;
